src/builtins.ts

```diff
@@ -771,6 +802,51 @@ export function compileCall(
         }
       }
     }
+    case BuiltinSymbols.nameof: {
+      // Check to make sure a parameter or a type was passed to the builtin
+      let resultType = evaluateConstantType(compiler, typeArguments, operands, reportNode);
+      if (resultType === null) {
+        compiler.currentType = compiler.program.stringInstance.type;
+        compiler.error(
+          DiagnosticCode.Operation_not_supported,
+          reportNode.typeArgumentsRange
+        );
+        return module.unreachable();
+      }
+
+      let value: string;
+      if (resultType.is(TypeFlags.REFERENCE)) {
+        if (resultType.classReference !== null) {
```

Comment:
Perhaps cache `classReference`

Reply:
I don't understand what this means.

Reply:
Sorry, what I meant was cache it in a local, like so

```ts
let classReference = resultType.classReference;
if (classReference) {
  value = classReference.name;
} else {
  assert(resultType.signatureReference);
  value = "Function";
}
```

More a nit with questionable benefit, but that's what other code tries to do.

Reply:
Ah. That makes sense. Thanks for clarifying. I don't mind following your code style. It's just that the style isn't documented, I have to infer it perfectly or I get lots of nit comments like this.

Reply:
All set.